pkg/dotc1z/sql_helpers.go

```diff
@@ -74,6 +78,33 @@ func (c *C1File) listConnectorObjects(ctx context.Context, tableName string, req
 		return nil, "", fmt.Errorf("c1file: invalid list request")
 	}
 
+	reqAnnos := annotations.Annotations(listReq.GetAnnotations())
+
+	var reqSyncID string
+	syncDetails := &c1zpb.SyncDetails{}
+	hasSyncIdAnno, err := reqAnnos.Pick(syncDetails)
+	if err != nil {
+		return nil, "", fmt.Errorf("c1file: failed to get sync id annotation: %w", err)
+	}
+
+	switch {
+	// If the request has a sync id annotation, use that
+	case hasSyncIdAnno && syncDetails.GetId() != "":
+		reqSyncID = syncDetails.GetId()
+
+	// We are currently syncing, so use the current sync id
+	case c.currentSyncID != "":
+		reqSyncID = c.currentSyncID
+
+	// We are viewing a sync, so use the view sync id
+	case c.viewSyncID != "":
```

Comment:
I think it might be worth dropping `viewSyncID` entirely now that we can specify which sync we care about as part of the request.